documentation/stackoverflow.workspace.adoc

```diff
@@ -3,211 +3,336 @@
-
-CALL apoc.load.json(url) YIELD value
-CALL apoc.util.sleep(250)  // careful with throttling
-
-UNWIND value.items AS q
-
-// create the questions
-MERGE (question:Question {uuid:q.question_id})
-  ON CREATE SET question.title = q.title, 
-  	question.link = q.share_link, 
-  	question.creation_date = q.creation_date, 
-  	question.accepted_answer_id=q.accepted_answer_id, 
-  	question.view_count=q.view_count,
-   	question.answer_count=q.answer_count,
-   	question.body_markdown=q.body_markdown
-
-// who asked the question
-MERGE (owner:User {uuid:coalesce(q.owner.user_id,'deleted')})
-  ON CREATE SET owner.display_name = q.owner.display_name
-MERGE (owner)-[:ASKED]->(question)
-
-// what tags do the questions have
-FOREACH (tagName IN q.tags | 
-  MERGE (tag:Tag {name:tagName}) 
-    ON CREATE SET tag.link = "https://stackoverflow.com/questions/tagged/" + tag.name
-  MERGE (question)-[:TAGGED]->(tag))
-
-// who answered the questions?
-FOREACH (a IN q.answers |
-   MERGE (question)<-[:ANSWERED]-(answer:Answer {uuid:a.answer_id})
-    ON CREATE SET answer.is_accepted = a.is_accepted,
-    answer.link=a.share_link,
-    answer.title=a.title,
-    answer.body_markdown=a.body_markdown,
-    answer.score=a.score,
-   	answer.favorite_score=a.favorite_score,
-   	answer.view_count=a.view_count
-   MERGE (answerer:User {uuid:coalesce(a.owner.user_id,'deleted')}) 
-    ON CREATE SET answerer.display_name = a.owner.display_name
-   MERGE (answer)<-[:PROVIDED]-(answerer)
-)
+MATCH (u:User)-[ ]->( )
+RETURN u,
+----
+====
 
-// who commented ont he question
-FOREACH (c in q.comments |
-  MERGE (question)<-[:COMMENTED_ON]-(comment:Comment {uuid:c.comment_id})
-    ON CREATE SET comment.link=c.link, comment.score=c.score
-  MERGE (commenter:User {uuid:coalesce(c.owner.user_id,'deleted')}) 
-    ON CREATE SET commenter.display_name = c.owner.display_name
-  MERGE (comment)<-[:COMMENTED]-(commenter)
-);
+[%collapsible]
+.Reveal the solution
+====
+[source,cypher]
 ----
+MATCH (u:User)-[p:PROVIDED]->(a:Answer)
+RETURN u,p,a
+----
+====
+
+The query will return this graph of `User` and `Answer` nodes.
+
+image::{img}/user-provided-answer.png[A graph of User and Answer nodes. There are clear clusters of users and answers.]
 
-Read More: https://neo4j.com/docs/apoc/current/import/load-json/#load-json-examples-stackoverflow[Import from StackOverflow API^]
+The distribution of nodes in the graph is not uniform - you can see that a few users have provided many answers.
 
-== Basic Queries
+In the next section, you will write Cypher to discover who those users are.
 
-Labels in the graph and counts for each:
+== Exploring Users
+
+As well as returning nodes, you can also return properties of nodes and aggregations (e.g. counts).
 
+.How many questions did users ask?
 [source,cypher]
 ----
-MATCH (n)
-RETURN labels(n) as label, count(*) as freq
-ORDER BY freq DESC;
+MATCH (u:User)-[:ASKED]->(q:Question)
```

Comment:
should we go with the new syntax?

```
MATCH (u:User)
RETURN u, count { (u)-[:ASKED]->() } as questions
ORDER BY questions DESC LIMIT 5

```

Reply:
I would always go with keeping it simple. Plus as you noted later it leads into another query.